src/serac/physics/materials/parameterized_solid_material.hpp

```diff
@@ -105,4 +105,95 @@ struct ParameterizedNeoHookeanSolid {
   double G0;       ///< base shear modulus
 };
 
+/**
+ * @brief Infers type resulting from algebraic expressions of a group of variables
+ *
+ * Useful if one needs to create a variable that is dual-valued if any operands are dual.
+ */
+template <typename... T>
+struct underlying_scalar {
```

Comment:
`scalar_sum` maybe? This name does not state that it is the type of the sum of the inputs, so it could be misleading.

Reply:
I agree this is not well named. `scalar_sum` more narrowly describes *how* it works,  but it doesn't express *why* you would use it. We could have used multiplication, for instance, instead of addition operators. The point is that if any of them were dual number-valued, the resulting type would be dual number-valued. I'm definitely open to suggestions on this. Maybe it will go away with one of Sam's suggested refactoring options.

Reply:
I'd say `scalar_type`, but then it would be `scalar_type::type`, so who knows. This is a temporary work around, so I'm ok with whatever as long as it's documented.

Reply:
Ok, I'll leave it for now.